src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -310,83 +310,95 @@ protected function decodeXrefStream($pdfData, $startxref, $xref = [])
 
         // decode data
         if ($valid_crs && isset($xrefcrs[1][3][0])) {
-            // number of bytes in a row
-            $rowlen = ($columns + 1);
-            // convert the stream into an array of integers
-            $sdata = unpack('C*', $xrefcrs[1][3][0]);
-            // split the rows
-            $sdata = array_chunk($sdata, $rowlen);
-            // initialize decoded array
-            $ddata = [];
-            // initialize first row with zeros
-            $prev_row = array_fill(0, $rowlen, 0);
-            // for each row apply PNG unpredictor
-            foreach ($sdata as $k => $row) {
-                // initialize new row
-                $ddata[$k] = [];
-                // get PNG predictor value
-                $predictor = (10 + $row[0]);
-                // for each byte on the row
-                for ($i = 1; $i <= $columns; ++$i) {
-                    // new index
-                    $j = ($i - 1);
-                    $row_up = $prev_row[$j];
-                    if (1 == $i) {
-                        $row_left = 0;
-                        $row_upleft = 0;
-                    } else {
-                        $row_left = $row[($i - 1)];
-                        $row_upleft = $prev_row[($j - 1)];
-                    }
-                    switch ($predictor) {
-                        case 10:  // PNG prediction (on encoding, PNG None on all rows)
-                            $ddata[$k][$j] = $row[$i];
-                            break;
+            if (isset($predictor)) {
```

Comment:
Is `isset` the right choice here? `$predictor`, when set, is of type `integer` based on:

https://github.com/smalot/pdfparser/blob/6db6fc934575070bcbf2821a656cac71c054deea/src/Smalot/PdfParser/RawData/RawDataParser.php#L291

For better code readability, I propose something like the following instead:

```
if (0 <= $predictor) { 
```

Maybe set `$predictor = null;` at the beginning of function too.

Reply:
Fixed